docs(animations): improve the aio animations introduction

dario-piotrowicz opened the pull request:
slight improvements to the angular.io animation introduction guide

URLs: 
 - https://angular.io/guide/animations
 - https://angular.io/guide/transition-and-triggers
 - https://angular.io/guide/complex-animation-sequences
 - https://angular.io/guide/reusable-animations
 - https://angular.io/guide/route-animations

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:


Issue Number: N/A


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


## Other information
I appreciate that some of these changes may be based on personal preferences/opinions, I'll naturally remove anything you don't agree with :slightly_smiling_face: 
( but I do believe that some in here are objective improvements :slightly_smiling_face: )

I'll also add comments in the PR for not self-explanatory changes :slightly_smiling_face:

dario-piotrowicz commented on aio/content/examples/animations/src/app/hero-list-page.component.html:
```diff
@@ -1,16 +1,13 @@
 <!-- #docplaster -->
 <h2>Filter/Stagger</h2>
 
-<form>
```
the form is not used in the example, so I think it should remove to keep the examples as simple as they can be

dario-piotrowicz commented on aio/content/examples/animations/src/app/hero-list-page.component.ts:
```diff
@@ -17,10 +17,11 @@ import { Hero } from './hero';
 // #enddocregion filter-animations
     trigger('pageAnimations', [
       transition(':enter', [
-        query('.hero, form', [
```
the form is not really related to the list we want to animate

dario-piotrowicz commented on aio/content/examples/animations/src/app/hero-list-page.component.ts:
```diff
@@ -17,10 +17,11 @@ import { Hero } from './hero';
           style({opacity: 0, transform: 'translateY(-100px)'}),
-          stagger(-30, [
-            animate('500ms cubic-bezier(0.35, 0, 0.25, 1)', style({ opacity: 1, transform: 'none' }))
+          stagger(30, [
```
The guide specifies that the elements enter from top to bottom but if the stagger value is negative they are inserted from bottom to top

[stackblitz example](https://stackblitz.com/edit/angular-anim-stagger-positive-vs-negative?file=src%2Fapp%2Fheroes%2Fheroes.component.ts)

dario-piotrowicz commented on aio/content/examples/animations/src/app/hero-list-page.component.ts:
```diff
@@ -32,7 +33,7 @@ import { Hero } from './hero';
       transition(':increment', [
         query(':enter', [
-          style({ opacity: 0, width: '0px' }),
+          style({ opacity: 0, width: 0 }),
```
here and below I just thought that `0` looks nicer and also shows that `width`'s value does not necessarily need to be a string

dario-piotrowicz commented on aio/content/examples/animations/src/app/open-close.component.4.ts:
```diff
@@ -22,7 +22,7 @@ import { trigger, transition, state, animate, style } from '@angular/animations'
         width: '100px',
         opacity: 0.8,
-        backgroundColor: '#c6ecff'
+        backgroundColor: 'blue'
```
above we have `'yellow'` but here we have a hex instead, I think it would be nicer the have the two consistent

dario-piotrowicz commented on aio/content/examples/animations/src/app/open-close.component.ts:
```diff
@@ -21,7 +21,7 @@ import { trigger, transition, state, animate, style, AnimationEvent } from '@ang
         height: '100px',
         opacity: 0.8,
-        backgroundColor: '#c6ecff'
+        backgroundColor: 'blue'
```
above we have `'yellow'` but here we have a hex instead, I think it would be nicer the have the two consistent

dario-piotrowicz commented on aio/content/examples/animations/src/app/hero-list-page.component.ts:
```diff
@@ -56,8 +56,8 @@ export class HeroListPageComponent implements OnInit {
 
 // #docregion filter-animations
-  heroTotal = -1;
-// #enddocregion filter-animations
```
the example uses the variable so I think it makes sense to include how the value is computed in it

dario-piotrowicz commented on aio/content/guide/animations.md:
```diff
@@ -263,7 +263,7 @@ What it does
 
 <tr>
-<td><code><a href="api/animations/state" class="code-anchor">state()</a></code></td>
+<td><code><a href="api/animations/state" class="code-anchor">state</a>()</code></td>
```
all standard method references in the docs don't have the brackets included in the link

dario-piotrowicz commented on aio/content/guide/complex-animation-sequences.md:
```diff
@@ -11,11 +11,11 @@ A basic understanding of the following concepts:
 * `query()` finds one or more inner HTML elements.
 * `stagger()` applies a cascading delay to animations for multiple elements.
-* [`group()`](api/animations/group) runs multiple animation steps in parallel.
+* <code><a href="api/animations/group" class="code-anchor">group</a>()</code> runs multiple animation steps in parallel.
```
here and in other places I make this change, it makes for an uglier markdown but a much nicer and more consistent outcome

before:
![Peek 2021-07-17 18-17](https://user-images.githubusercontent.com/61631103/126044847-928dd5c4-d484-492b-8d86-e11b89263356.gif)

after:
![Peek 2021-07-17 18-18](https://user-images.githubusercontent.com/61631103/126044852-22c40f78-1990-47e7-a111-46f81742efc5.gif)

dario-piotrowicz commented on aio/content/guide/complex-animation-sequences.md:
```diff
@@ -26,13 +26,11 @@ The `query()` function allows you to find inner elements within the element that
 * Use `query()` to look for an element entering the page that meets certain criteria.
 
-* For each of these elements, use `style()` to set the same initial style for the element. Make it invisible and use `transform` to move it out of position so that it can slide into place.
+* For each of these elements, use `style()` to set the same initial style for the element. Make it transparent and use `transform` to move it out of position so that it can slide into place.
```
not sure about the word `invisible` makes me think of the css visibility property...

I think "transparent" is actually a better word here (it is the synonym of opaque)

dario-piotrowicz commented on aio/content/guide/complex-animation-sequences.md:
```diff
@@ -42,20 +40,22 @@ The following example demonstrates how to use `query()` and `stagger()` function
 
-In the following example, using groups on both `:enter` and `:leave` allow for two different timing configurations. They're applied to the same element in parallel, but run independently.
+The following example, uses <code><a href="api/animations/group" class="code-anchor">group</a></code>s on both `:enter` and `:leave` for two different timing configurations, thus applying two independent animations to the same element in parallel.
+<!-- I find this example a bit confusing, I think you be better to have a simpler example with just one group instead of
+making things more complex -->
```
sorry, regardless on my typo I put this here for myself as a reminder I will remove it

but yeah I do believe that the example here is unnecessarily complex and that it could be simpler

dario-piotrowicz commented on aio/content/guide/complex-animation-sequences.md:
```diff
@@ -64,35 +64,35 @@ A second function called `sequence()` lets you run those same animations one aft
 ## Filter animation example
 
-Let's take a look at another animation on the live example page. Under the Filter/Stagger tab, enter some text into the **Search Heroes** text box, such as `Magnet` or `tornado`.
+Let's take a look at another animation example. We have a search input which filters some heroes.
```
I have no idea what live example page this is mentioning (I mean, I've found it in the code, but I could not find it in the docs)

aikidave replied:
Hm. Are we missing a link? If so, let's add it! Can you tell me where the example is?

dario-piotrowicz replied:
Sorry, it's been a couple of weeks since I made this change so I am not 100% sure, but I think that the issue is that here it mentions this example but it's not like it's introducing it and I could not find anywhere were it was actually uses, so I think that my understanding was that probably such an example existed but it was removed or something but still referenced here

but the code is actually still there, so it may actually be used somewhere....

so yeah I am not completely sure :sweat_smile: , I'll need to have another look at it and give you more details, sorry :pray: :bow:

dario-piotrowicz replied:
ok.... so sorry I was mistaken! :bow: 

Here's the example: https://angular.io/guide/example-apps-list#animations

The thing is that that is the only single place in the whole site which actually links to the example....

I don't think we should just here say something and link the page I just mentioned, I think we should mention the live example somewhere (like at the start of the guides) and then refer to that :thinking: 

What do you think @aikidave do you agree?

dario-piotrowicz replied:
@aikidave I am so sorry :bow: :bow: :bow: :bow: :bow: :bow: :bow: :bow: :bow: :bow: 

The guide already presents the example:
![Screenshot at 2021-09-01 23-09-59](https://user-images.githubusercontent.com/61631103/131752045-16f998ee-9bcf-47e5-9aaa-b45580e972c4.png)

(I missed that because if just has it as `<live-example></live-example>` I didn't know that was another way to include it :sweat: )

So should I just revert the sentence to its original form? :thinking: 

> ps: I admit that I may just have been stupid... but if I got confused/lost about this I believe it is possible that it is not really crystal clear... I think my issue was that this is just quickly mentioned like this at the beginning of the guide, and then mentioned again only a couple of times many sections/guides later... I wonder if we can do anything about that :sweat:

dario-piotrowicz replied:
Sorry I am not sure if your suggestion has a typo of something (I don't get the `in the .` part :sweat:)

___

Anyways my issue here is a general one, seems like we have this live example and I actually I think that all (or most of) the examples presented in this guide are from the live example.

If you're ok with it I am thinking to just revert it back to the original text (as it is sort of right, it's just that I didn't get it) and then if we want I can go back to the guide and try to make generally clear that the examples are taken from the live/example one.

What do you say about that? :slightly_smiling_face:

dario-piotrowicz replied:
Actually this was involved into a merge conflict so I too the occasion to revert it back to it's original form, @aikidave please let me know if it's ok :sweat:

aikidave replied:
I have a plan to do some additional work on these topics; your efforts are making that work much easier, @dario-piotrowicz !

As the merge conflict caused you to revert these sentences back to their original form, I'm fine if we leave them as is. Let's not have perfect be the enemy of good!

dario-piotrowicz replied:
wow I am really glad if I could be any useful!!!! :smile: 

Yeah the merge conflict was nothing breaking really, but since I was already thinking of reverting it back I just took it as a good chance to just do that :sweat_smile: 

Any yeah I agree with the perfection thing, this PR's been open for ages, the priority should be hopefully to get it merged (as it is improving things anyways) and take care of other stuff later on :slightly_smiling_face: 

> PS: if there'll be anything I could do to assist you please let me know! :smile:

dario-piotrowicz commented on aio/content/guide/complex-animation-sequences.md:
```diff
@@ -64,35 +64,35 @@ A second function called `sequence()` lets you run those same animations one aft
 
 <code-example path="animations/src/app/hero-list-page.component.ts" header="src/app/hero-list-page.component.ts" region="filter-animations" language="typescript"></code-example>
 
-The animation does the following:
```
This was not the best wording I believe, for example we had something like:

- The animation does the following:
 * Ignores any animations ...
 
doesn't really make too much sense, does it? :confused:

aikidave replied:
What about:

"The code in this example performs the following tasks:" ?

dario-piotrowicz replied:
yeah it makes much more sense than what I put there :sweat_smile:

dario-piotrowicz commented on aio/content/guide/reusable-animations.md:
```diff
@@ -11,11 +11,11 @@ Before continuing with this topic, you should be familiar with the following:
 <code-example path="animations/src/app/animations.1.ts" header="src/app/animations.ts" region="animation-const" language="typescript"></code-example>
 
-In the above code snippet, `transAnimation` is made reusable by declaring it as an export variable.
+In the above code snippet, `transitionAnimation` is made reusable by declaring it as an export variable.
```
this was inconsistent, in the code the variable is actually called `transitionAnimation`:
![Screenshot at 2021-07-17 17-50-54](https://user-images.githubusercontent.com/61631103/126050411-9a8bfdd5-9705-48a9-8d29-c79135316cf6.png)

dario-piotrowicz commented on aio/content/guide/route-animations.md:
```diff
@@ -78,9 +78,9 @@ The following code snippet defines a reusable animation named `slideInAnimation`
 <code-example path="animations/src/app/animations.ts" header="src/app/animations.ts" region="route-animations" language="typescript"></code-example>
 
-The animation definition does several things:
+The animation definition (which we will break down in the next sections) does several things:
```
here and below I wanted to emphasise that the next sections break down the same example

In the guide ([link](http://localhost:4200/guide/route-animations#animation-definition)) that's not made clear and got me quite confused while reading

(I wasn't sure if it was adding or improving the example or something)

dario-piotrowicz commented on aio/content/guide/transition-and-triggers.md:
```diff
@@ -33,7 +33,7 @@ Use a double arrow syntax to specify state-to-state transitions in both directio
 
 In the two-state button example, the wildcard isn't that useful because there are only two possible states, `open` and `closed`.
-Wildcard states are better when an element in one particular state has multiple potential states that it can change to.
+Wildcard states are more useful when an element in one particular state has multiple potential states that it can change to.
```
better seems a pretty ambiguous term here, how are they better? do they change?

aikidave replied:
Maybe:

`Wildcard state can help you when an element...`

aikidave replied:
How about: 

"Use wildcard states when an element in one particular state has multiple potential states that it can change to."

Puts more of the focus on the user than the feature.

dario-piotrowicz replied:
The problem is that your latest sentence doesn't follow what the previous sentence says,
how about: 
```
In general you use wildcard states when an element in one particular state has multiple potential states that it can change to.
```

so that we get:
```
In the two-state button example, the wildcard isn't that useful because there are only two possible states, `open` and `closed`.
In general you use wildcard states when an element in one particular state has multiple potential states that it can change to.
```

?

aikidave replied:
Sounds good. Just one small change:

`In general, use wildcard states when an element in one particular state has multiple potential states that it can change to.`

dario-piotrowicz commented on aio/content/guide/transition-and-triggers.md:
```diff
@@ -200,7 +200,7 @@ The animation `trigger()` function emits *callbacks* when it starts and when it
 <code-example path="animations/src/app/open-close.component.ts" header="src/app/open-close.component.ts" region="events1" language="typescript"></code-example>
 
-In the HTML template, the animation event is passed back via `$event`, as `@trigger.start` and `@trigger.done`, where `trigger` is the name of the trigger being used.
+In the HTML template, the animation event is passed back via `$event`, as `@triggerName.start` and `@triggerName.done`, where `triggerName` is the name of the trigger being used.
```
if we used `trigger` as the name, the word would be considered a link to the animation trigger function is just wrong and confusing:
![Screenshot at 2021-07-17 23-44-07](https://user-images.githubusercontent.com/61631103/126050962-24d81212-69af-480d-96d8-a6c22804d94b.png)

___

Besides that, we already use the term `triggerName` in the first page of the guide ([link](http://localhost:4200/guide/animations#defining-animations-and-attaching-them-to-the-html-template)):
![Screenshot at 2021-07-17 23-45-18](https://user-images.githubusercontent.com/61631103/126050989-783ae6a7-d1dd-4088-963c-72901ab07d44.png)
so using it here is also more consistent

dario-piotrowicz commented:
@petebacondarwin  so sorry for bothering you :bow: , I was just wondering if you could help me out getting a pair of eyes on this PR as I think I am making a few nice improvements to the docs but I'm afraid this PR is just getting forgotten :cry:

petebacondarwin commented:
I'm going to ping @aikidave on this one to get some review from the docs team.

dario-piotrowicz commented:
> I'm going to ping @aikidave on this one to get some review from the docs team.

Thank you so very much!!! :slightly_smiling_face: :slightly_smiling_face: :slightly_smiling_face:

aikidave commented:
I'm on this! Stay tuned, Dario! Won't keep you waiting!

On Fri, Aug 20, 2021 at 2:04 PM Dario Piotrowicz ***@***.***>
wrote:

> I'm going to ping @aikidave <https://github.com/aikidave> on this one to
> get some review from the docs team.
>
> Thank you so very much!!! 🙂 🙂 🙂
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular/pull/42885#issuecomment-902956383>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHGDHY54EINELB3W42WKEDTT527OJANCNFSM5ARHLDXA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>

dario-piotrowicz commented:
> I'm on this! Stay tuned, Dario! Won't keep you waiting!
> […](#)

@aikidave  Thanks!!!!! :heart_eyes: :heart_eyes: :heart_eyes:

aikidave commented on aio/content/guide/animations.md:
```diff
@@ -84,14 +84,14 @@ In HTML, these attributes are set using ordinary CSS styles such as color and op
 Use Angular's `state()` function to define different states to call at the end of each transition. This function takes two arguments: a unique name like `open` or `closed` and a `style()` function.
 
-Use the `style()` function to define a set of styles to associate with a given state name. Note that the style attributes must be in [*camelCase*](guide/glossary#case-conventions).
+Use the `style()` function to define a set of styles to associate with a given state name. Note that the style attributes containing dashed must either be written in [*camelCase*](guide/glossary#case-conventions) (e.g. `'backgroundColor'`) or denoted as strings (e.g. `'background-color'`).
```
```suggestion
Use the `style()` function to define a set of styles to associate with a given state name. You must write style attributes that contain dashes in [*camelCase*](guide/glossary#case-conventions), such as `'backgroundColor'` or denoted as strings, such as `'background-color'`.
```

aikidave commented on aio/content/guide/complex-animation-sequences.md:
```diff
@@ -42,20 +40,22 @@ The following example demonstrates how to use `query()` and `stagger()` function
 ## Parallel animation using group() function
 
-You've seen how to add a delay between each successive animation. But you may also want to configure animations that happen in parallel. For example, you may want to animate two CSS properties of the same element but use a different `easing` function for each one. For this, you can use the animation [`group()`](api/animations/group) function.
+You've seen how to add a delay between each successive animation. But you may also want to configure animations that happen in parallel. For example, you may want to animate two CSS properties of the same element but use a different `easing` function for each one. For this, you can use the animation <code><a href="api/animations/group" class="code-anchor">group</a>()</code> function.
```
```suggestion
Previous sections describe how to add a delay between successive animations. But you might also want to configure animations that happen in parallel. For example, you might want to animate two CSS properties of the same element but use a different `easing` function for each one. To accomplish this task, you can use the animation <code><a href="api/animations/group" class="code-anchor">group</a>()</code> function.
```

aikidave commented on aio/content/guide/complex-animation-sequences.md:
```diff
@@ -64,35 +64,35 @@ A second function called `sequence()` lets you run those same animations one aft
+Angular does the following:
 
-* Ignores any animations that are performed when the user first opens or navigates to this page. The filter narrows what is already there, so it assumes that any HTML elements to be animated already exist in the DOM.
+* Doesn't animate any element when the user first opens or navigates to this page (the filter animation narrows what is already there, so it only works on elements already exist in the DOM).
```
```suggestion
* Skips animating elements when the user first opens or navigates to this page (the filter animation narrows what is already there, so it only works on elements that already exist in the DOM).
```

What do you think of this wording?

aikidave commented on aio/content/guide/route-animations.md:
```diff
@@ -21,7 +21,7 @@ To enable routing transition animation, do the following:
 
 
-Let's illustrate a router transition animation by navigating between two routes, *Home* and *About* associated with the `HomeComponent` and `AboutComponent` views respectively. Both of these component views are children of the top-most view, hosted by `AppComponent`. We'll implement a router transition animation that slides in the new view to the right and slides out the old view when the user na ...
+Let's illustrate a router transition animation by navigating between two routes, *Home* and *About* associated with the `HomeComponent` and `AboutComponent` views respectively. Both of these component views are children of the top-most view, hosted by `AppComponent`. We'll implement a router transition animation that slides in the new view from the left and slides out to the right the old view wh ...
```
```suggestion
Let's illustrate a router transition animation by navigating between two routes, *Home* and *About*, which are associated with the `HomeComponent` and `AboutComponent` views respectively. Both of these component views are children of the top-most view, hosted by `AppComponent`. We'll implement a router transition animation that slides in the new view from the left and slides out to the right the old view when the user navigates between the two.
```

aikidave commented on aio/content/guide/route-animations.md:
```diff
@@ -33,11 +33,11 @@ Let's illustrate a router transition animation by navigating between two routes,
 To begin, configure a set of routes using methods available in the `RouterModule` class. This route configuration tells the router how to navigate.
 
-Use the `RouterModule.forRoot` method to define a set of routes. Also, import this `RouterModule` to the `imports` array of the main module, `AppModule`.
+Use the `RouterModule.forRoot` method to define a set of routes. Also, import this `RouterModule` by adding it to the `imports` array of the main module, `AppModule`.
```
```suggestion
Use the `RouterModule.forRoot` method to define a set of routes. Also, add `RouterModule` to the `imports` array of the main module, `AppModule`.
```

Just tightening up the wording here.

aikidave commented on aio/content/guide/route-animations.md:
```diff
@@ -33,11 +33,11 @@ Let's illustrate a router transition animation by navigating between two routes,
 <div class="alert is-helpful">
 
-**Note:** Use the `RouterModule.forRoot` method in the root module, `AppModule`, to register top-level application routes and providers. For feature modules, call the `RouterModule.forChild` method to register additional routes.
+**Note:** Use the `RouterModule.forRoot` method in the root module, `AppModule`, to register top-level application routes and providers. For feature modules, call the  `RouterModule.forChild` method instead.
```
```suggestion
**Note:** Use the `RouterModule.forRoot` method in the root module, `AppModule`, to register top-level application routes and providers. For feature modules, call the `RouterModule.forChild` method instead.
```

aikidave commented on aio/content/guide/route-animations.md:
```diff
@@ -57,9 +57,9 @@ In addition to `path` and `component`, the `data` property of each route defines
 ## Router outlet
 
-After configuring the routes, tell the Angular router where to render the views when matched with a route. You can set a router outlet by inserting a `<router-outlet>` container inside the root `AppComponent` template.
+After configuring the routes, tell the Angular router where to render the views when matched with a route. You can set a router outlet by inserting a `<router-outlet>` directive inside the root `AppComponent` template.
```
```suggestion
After configuring the routes, add a `<router-outlet>` to configure where to render the views when matched with a route. You can set a router outlet by inserting a `<router-outlet>` directive inside the root `AppComponent` template.
```

dario-piotrowicz replied:
mh.... are you sure? :thinking: 

aren't we mentioning `<router-outlet>` too often here? 

like "add a router-outlet..." then just in the immediately next sentence "...you can set a router outlet by inserting a <router-outlet>"

dario-piotrowicz replied:
Actually... if someone is reading the animations guide... I guess we could assume that they do know how to use the router already... maybe we could just get rid of the second sentence altogether? 

or just cpmbine the two like:
```
After configuring the routes, add a `<router-outlet>` in the root `AppComponent`
template to configure where to render the views when matched with a route.
```
?

aikidave replied:
How about:

After configuring the routes, add a `<router-outlet>` inside the root `AppComponent` template. The `<router-outlet>` tells the Angular router where to render the views when matched with a route.

aikidave commented on aio/content/guide/transition-and-triggers.md:
```diff
@@ -81,7 +81,7 @@ This section shows how to animate elements entering or leaving a page.
 </div>
 
-Now add a new behavior:
+Let's now add a new behavior:
```
```suggestion
Add a new behavior:
```

aikidave commented on aio/content/guide/transition-and-triggers.md:
```diff
@@ -127,7 +127,7 @@ The `transition()` function takes additional selector values, `:increment` and `
 <div class="alert is-helpful">
 
-**Note:** The following example uses `query()` and `stagger()` methods, which is discussed in the [complex sequences](guide/complex-animation-sequences#complex-sequence) page.
+**Note:** The following example uses `query()` and `stagger()` methods, which are discussed in the [complex sequences](guide/complex-animation-sequences#complex-sequence) page.
```
```suggestion
**Note:** The following example uses `query()` and `stagger()` methods. For more information on these methods, see the [complex sequences](guide/complex-animation-sequences#complex-sequence) page.
```

aikidave commented on aio/content/guide/transition-and-triggers.md:
```diff
@@ -140,9 +140,9 @@ If a trigger contains a boolean value as a binding value, then this value can be
 </code-example>
 
-In the code snippet above, the HTML template binds a `<div>` element to a trigger named `openClose` with a status expression of `isOpen`, and with possible values of `true` and `false`. This is an alternative to the practice of creating two named states of `open` and `close`.
+In the code snippet above, the HTML template binds a `<div>` element to a trigger named `openClose` with a status expression of `isOpen`, and with possible values of `true` and `false`. This is an alternative to the practice of creating two named states like `open` and `close`.
```
```suggestion
In the code snippet above, the HTML template binds a `<div>` element to a trigger named `openClose` with a status expression of `isOpen`, and with possible values of `true` and `false`. This pattern is an alternative to the practice of creating two named states like `open` and `close`.
```

aikidave commented on aio/content/guide/transition-and-triggers.md:
```diff
@@ -140,9 +140,9 @@ If a trigger contains a boolean value as a binding value, then this value can be
+In the code snippet above, the HTML template binds a `<div>` element to a trigger named `openClose` with a status expression of `isOpen`, and with possible values of `true` and `false`. This is an alternative to the practice of creating two named states like `open` and `close`.
 
-In the component code, in the `@Component` metadata under the `animations:` property, when the state evaluates to `true` (meaning "open" here), the associated HTML element's height is a wildcard style or default. In this case, use whatever height the element already had before the animation started. When the element is "closed," the element animates to a height of 0, which makes it invisible.
+In the component code, inside the `@Component` metadata under the `animations:` property, when the state evaluates to `true` (meaning "open" here), the associated HTML element's height is a wildcard style or default. In this case, the animation uses whatever height the element already had before the animation started. When the element is "closed", it gets animated to a height of 0, which makes it ...
```
```suggestion
In the component code, inside the `@Component` metadata under the `animations:` property, when the state evaluates to `true` (meaning "open" here), the associated HTML element's height is a wildcard style or default. In this case, the animation uses whatever height the element already had before the animation started. When the element is "closed", the element gets animated to a height of 0, which makes it invisible.
```

aikidave commented on aio/content/guide/transition-and-triggers.md:
```diff
@@ -181,7 +181,7 @@ However, selective child animations can still be run on a disabled parent in one
 Those elements can still animate.
 
-* A subanimation can be queried by a parent and then later animated with the `animateChild()` function.
+* A sub-animation can be queried by a parent and then later animated with the `animateChild()` function.
```
```suggestion
* A sub-animation can be queried by a parent and then later animated with the `animateChild()` function.
```

Do we need a new term, sub-animation? Could we just refer to this as a secondary animation or a child animation?

dario-piotrowicz replied:
mh... I'd imagine that we could go with `secondary animation`, it does sound better, doesn't it? :slightly_smiling_face: 

I guess I would personally not go with a `child animation` since that can be more open to interpretation and/or be more confusing (like that to me could sound like a specific concept in the angular animations package or something... :thinking: )

aikidave replied:
In general, I prefer to avoid introducing new terms. I think `child animation` works, because you already refer to the parent animation in the previous paragraph.

dario-piotrowicz replied:
mh... the more I am thinking of it the more "child animation" is actually growing on me :sweat_smile::+1:

aikithoughts reviewed (commented):
Wow. There's a lot here, @dario-piotrowicz. I appreciate all the thought you've clearly put into this.

I've made some wording suggestions to help make the meaning a little clearer. Reviewing this PR, however, makes me think that we could do even more to make this content easier for developers to read. 

Your PR is a great step forward in that effort! Thank you!

dario-piotrowicz commented:
> Wow. There's a lot here, @dario-piotrowicz. I appreciate all the thought you've clearly put into this.
> 
> I've made some wording suggestions to help make the meaning a little clearer. Reviewing this PR, however, makes me think that we could do even more to make this content easier for developers to read.
> 
> Your PR is a great step forward in that effort! Thank you!

@aikidave Thank you so very much for spending your precious time reviewing my PR, yeah I can't imagine it taking little time, I really appreciate it :slightly_smiling_face: 

About the "even more to make the is content easier" I totally agree, I remember there were parts which were not just the easiest to follow (and not because of the topic), but ultimately I just figured it would be best not to make any major change

aikithoughts reviewed (commented):
Nice work! Added a few more comments to the conversation.

mary-poppins commented:
You can preview 9a7dfd1 at https://pr42885-9a7dfd1.ngbuilds.io/.
You can preview cc8d305 at https://pr42885-cc8d305.ngbuilds.io/.
You can preview 24702d5 at https://pr42885-24702d5.ngbuilds.io/.
You can preview 8ccca9d at https://pr42885-8ccca9d.ngbuilds.io/.
You can preview c88576d at https://pr42885-c88576d.ngbuilds.io/.
You can preview 41e244c at https://pr42885-41e244c.ngbuilds.io/.
You can preview d812ee9 at https://pr42885-d812ee9.ngbuilds.io/.
You can preview 212a848 at https://pr42885-212a848.ngbuilds.io/.

dario-piotrowicz commented on aio/content/guide/route-animations.md:
```diff
@@ -57,9 +57,9 @@ In addition to `path` and `component`, the `data` property of each route defines
+After configuring the routes, tell the Angular router where to render the views when matched with a route. You can set a router outlet by inserting a `<router-outlet>` directive inside the root `AppComponent` template.
 
-The `<router-outlet>` container has an attribute directive that contains data about active routes and their states, based on the `data` property that we set in the route configuration.
+The `<router-outlet>` directive has an attribute directive that contains data about active routes and their states, based on the `data` property that we set in the route configuration.
```
the router-outlet is a directive and not a container, that's the only thing I noticed and was fixing... but reading the sentence now it just seems off.... I'll need to think about this a bit :thinking:

mary-poppins commented:
You can preview 7b0f5f2 at https://pr42885-7b0f5f2.ngbuilds.io/.

mary-poppins commented:
You can preview ae56feb at https://pr42885-ae56feb.ngbuilds.io/.

aikithoughts reviewed (approved):
I think this is all a great improvement. LGTM!

Reviewed-for: global-docs-approvers

dario-piotrowicz commented on aio/content/guide/route-animations.md:
```diff
@@ -95,29 +95,31 @@ Make the animation definition available in your application by adding the reusab
 ### Styling the host and child components
 
-During a transition, a new view is inserted directly after the old one and both elements appear on screen at the same time. To prevent this, apply additional styling to the host view, and to the removed and inserted child views. The host view must use relative positioning, and the child views must use absolute positioning. Adding styling to the views animates the containers in place, without the  ...
+During a transition, a new view is inserted directly after the old one and both elements appear on screen at the same time. To prevent this, apply additional styling to the host view, and to the removed and inserted child views. For this the host view uses relative positioning, and the child views use absolute positioning. Adding styling to the views animates the containers in place, without the  ...
```
Yes your version is much more clear! :slightly_smiling_face::+1:

It is correct except that the child view needs to use `absolute` and not `relative` positioning :slightly_smiling_face::+1:

aikithoughts reviewed (approved):
From a documentation perspective, this looks good! Thanks, @dario-piotrowicz ! I'll leave code reviews to those who know more than I. :)

Reviewed-for: global-docs-approvers

mary-poppins commented:
You can preview 4aaffdf at https://pr42885-4aaffdf.ngbuilds.io/.

dario-piotrowicz commented:
Thank you so very much @aikidave :slightly_smiling_face: 

I addressed all your comments, the only thing we still need to sort out is this I believe: https://github.com/angular/angular/pull/42885/files#r710365959

:slightly_smiling_face::+1: 

(then afterwords maybe you could ping some team member to review this? :heart_eyes: )

mary-poppins commented:
You can preview 6ea6570 at https://pr42885-6ea6570.ngbuilds.io/.

dario-piotrowicz commented:
@petebacondarwin and @gkalpak so sorry for bothering you :cry:, @aikidave seems fine with the changes from a documentation point of view, could either of you have a look from a code point of view? :bow: :pray: (or ping someone that could? :slightly_smiling_face:)

gkalpak commented on aio/content/guide/complex-animation-sequences.md:
```diff
@@ -72,27 +72,27 @@ The HTML template contains a trigger called `filterAnimation`.
+The code in this example performs the following tasks:
 
-* Ignores any animations that are performed when the user first opens or navigates to this page. The filter narrows what is already there, so it assumes that any HTML elements to be animated already exist in the DOM.
+* Skips animating elements when the user first opens or navigates to this page (the filter animation narrows what is already there, so it only works on elements that already exist in the DOM).
```
"Skips animating elements" is a bit confusing here (at first I was interpreting it as "skips elements that are animated").
How about:

```suggestion
* Skips animations when the user first opens or navigates to this page (the filter animation narrows what is already there, so it only works on elements that already exist in the DOM).
```

dario-piotrowicz replied:
ah.... yeah I understand what you mean :+1: 

(like "animating" could interpreted as the verb for the execution or the adjective for the elements)

gkalpak commented on aio/content/examples/animations/src/app/app.component.ts:
```diff
@@ -23,7 +23,6 @@ import { slideInAnimation } from './animations';
 // #enddocregion decorator
     slideInAnimation
 // #docregion decorator
-    // animation triggers go here
```
OOC, why remove this?

dario-piotrowicz replied:
Actually now that you ask about it I am not really sure the removal is necessary... :thinking: 

The fact is that this appears in two places:

[![Screenshot at 2021-09-27 22-30-48](https://user-images.githubusercontent.com/61631103/134988609-104777c4-069f-4669-a44a-db21c35cd62d.png)](https://angular.io/guide/transition-and-triggers#disabling-all-animations)

[![Screenshot at 2021-09-27 22-32-18](https://user-images.githubusercontent.com/61631103/134988551-7a53477b-8697-47f4-bff8-31d5855ed260.png)](https://angular.io/guide/route-animations#animation-definition)

both cases are not indicating where the triggers should go (that is covered in other parts of the guide) and to me that comment sort of just looks distracting as there is already an animation in the array

so it is sort of like us telling the reader where the triggers go and then adding this comment sort of indicating where they go... not sure :sweat_smile: 

please let me know what you think, if you think it should stay I don't mind putting it back in :slightly_smiling_face:

gkalpak replied:
I see. In the two cases you mentioned it is indeed redundant (or even confusing).
But I think it is still relevant in [Adding the animation metadata property][1] section (which uses the `decorator` docregion).

How about we keep the comment only for the `decorator` docregion then?

[1]: https://angular.io/guide/animations#step-3-adding-the-animation-metadata-property

dario-piotrowicz replied:
yeah.... I am so sorry I got confused with the start and end of regions and thought it was only used in those 2 :sweat: 

(quite silly of me :sweat_smile:)

anyway yeah, sounds like a plan! :smiley::+1:

dario-piotrowicz replied:
done: https://github.com/angular/angular/pull/42885/commits/55de8e0038c5aec2e6b2315dfbd76b89844a1b7b

let me know if it's ok :smile: 

would you add a comma after `slideInAnimation`? :thinking:

gkalpak replied:
👌

gkalpak reviewed (changes requested):
Great work as usual, @dario-piotrowicz 💯
I've just left a few questions/suggestions.

dario-piotrowicz commented on aio/content/guide/complex-animation-sequences.md:
```diff
@@ -42,20 +40,22 @@ The following example demonstrates how to use `query()` and `stagger()` function
 
-In the following example, using groups on both `:enter` and `:leave` allow for two different timing configurations. They're applied to the same element in parallel, but run independently.
+The following example, uses <code><a href="api/animations/group" class="code-anchor">group</a></code>s on both `:enter` and `:leave` for two different timing configurations, thus applying two independent animations to the same element in parallel.
+<!-- I find this example a bit confusing, I think you be better to have a simpler example with just one group instead of
+making things more complex -->
```
sorry! gosh I can't believe this wasn't already removed :sweat: :bow: :pray: +

It's been ages since I created this PR, not even sure why I found this too complicated, we can ignore this for now I think :slightly_smiling_face:

dario-piotrowicz commented on aio/content/guide/animations.md:
```diff
@@ -84,14 +84,14 @@ In HTML, these attributes are set using ordinary CSS styles such as color and op
 Use Angular's `state()` function to define different states to call at the end of each transition. This function takes two arguments: a unique name like `open` or `closed` and a `style()` function.
 
-Use the `style()` function to define a set of styles to associate with a given state name. Note that the style attributes must be in [*camelCase*](guide/glossary#case-conventions).
+Use the `style()` function to define a set of styles to associate with a given state name. You must write style attributes that contain dashes in [*camelCase*](guide/glossary#case-conventions), such as `'backgroundColor'` or denoted as strings, such as `'background-color'`.
```
ah ok :slightly_smiling_face::+1: 

But I think something in your suggestion is missing... :thinking: 
`You must [*camelCase*](guide/glossary#case-conventions) for style attributes...`
did you missed something like "use"?
`You must use [*camelCase*](guide/glossary#case-conventions) for style attributes...`
or maybe the "for" is to be removed? (using camelCase as a verb)?
`You must [*camelCase*](guide/glossary#case-conventions) style attributes...`

gkalpak replied:
Sorry, I meant _"You must **use** camelCase for ..."_ 😇

dario-piotrowicz replied:
done :smile:

dario-piotrowicz commented on aio/content/guide/reusable-animations.md:
```diff
@@ -26,7 +26,7 @@ You can also export a part of an animation. For example, the following snippet e
 <code-example path="animations/src/app/animations.1.ts" header="src/app/animations.1.ts" region="trigger-const" language="typescript"></code-example>
 
-From this point, you can import resuable animation variables in your component class. For example, the following code snippet imports the `transAnimation` variable for use in the `useAnimation()` method.
+From this point, you can import reusable animation variables in your component class. For example, the following code snippet imports the `transitionAnimation` variable and uses it via the the `useAnimation()` function.
```
:man_facepalming::+1:

dario-piotrowicz commented on aio/content/guide/route-animations.md:
```diff
@@ -95,29 +95,31 @@ Make the animation definition available in your application by adding the reusab
 ### Styling the host and child components
 
-During a transition, a new view is inserted directly after the old one and both elements appear on screen at the same time. To prevent this, apply additional styling to the host view, and to the removed and inserted child views. The host view must use relative positioning, and the child views must use absolute positioning. Adding styling to the views animates the containers in place, without the  ...
+During a transition, a new view is inserted directly after the old one and both elements appear on screen at the same time. To prevent this behavior, update the host view to use relative positioning. Then, update the removed and inserted child views to use absolute positioning. Adding these styles to the views animates the containers in place, without manipulating the DOM.
```
ah ok! :+1: 

I think I misinterpreted the original sentence, yeah yours make much more sense :sweat_smile:

mary-poppins commented:
You can preview b45e48c at https://pr42885-b45e48c.ngbuilds.io/.

dario-piotrowicz commented on aio/content/guide/reusable-animations.md:
```diff
@@ -11,11 +11,11 @@ Before continuing with this topic, you should be familiar with the following:
 ## Creating reusable animations
 
-To create a reusable animation, use the [`animation()`](api/animations/animation) method to define an animation in a separate `.ts` file and declare this animation definition as a `const` export variable. You can then import and reuse this animation in any of your application components using the [`useAnimation()`](api/animations/useAnimation) API.
+To create a reusable animation, use the <code><a href="api/animations/animation" class="code-anchor">animation</a>()</code> function to define an animation in a separate `.ts` file and declare this animation definition as a `const` export variable. You can then import and reuse this animation in any of your application components using the <code><a href="api/animations/useAnimation" class="code-a ...
```
that works great!!! amazing suggestion!!! :heart_eyes: I'll amend the various links :+1:

dario-piotrowicz replied:
done :slightly_smiling_face:  (https://github.com/angular/angular/pull/42885/commits/7e66c5ce4a48536eabdb6d34587c83739b110e81)

mary-poppins commented:
You can preview 7e66c5c at https://pr42885-7e66c5c.ngbuilds.io/.

dario-piotrowicz commented on aio/content/guide/route-animations.md:
```diff
@@ -95,29 +95,31 @@ Make the animation definition available in your application by adding the reusab
+During a transition, a new view is inserted directly after the old one and both elements appear on screen at the same time. To prevent this behavior, update the host view to use relative positioning. Then, update the removed and inserted child views to use absolute positioning. Adding these styles to the views animates the containers in place, without manipulating the DOM.
 
-<code-example path="animations/src/app/animations.ts" header="src/app/animations.ts" region="style-view" language="typescript"></code-example>
+<code-example path="animations/src/app/animations.ts" header="src/app/animations.ts (Break down)" region="style-view" language="typescript"></code-example>
```
ah.... sorry if it's confusing :sweat: 

basically the same file is presented fully at the beginning of the section:
![Screenshot at 2021-09-27 22-21-05](https://user-images.githubusercontent.com/61631103/134987167-32a6bc1e-a82c-4e2f-bda4-1517a1828f10.png)

and then it's broken down two different smaller examples/explanations below:
![Screenshot at 2021-09-27 22-21-26](https://user-images.githubusercontent.com/61631103/134987291-582f36dd-86a9-40bd-a6e2-49da40c80762.png)
![Screenshot at 2021-09-27 22-23-26](https://user-images.githubusercontent.com/61631103/134987353-f4a079ff-174a-4518-a5db-64e751120542.png)

so I wanted to add something to sort of emphasize that this is just a broken down of the example shown above...

is this just silly unnecessary? :sweat:

gkalpak replied:
I see. That makes sense. We typically use ` (excerpt)` for that. Would that work here (to keep it consistent with the rest of the docs)?

dario-piotrowicz replied:
yeah that looks nicer :slightly_smiling_face::+1:

gkalpak commented on aio/content/guide/route-animations.md:
```diff
@@ -95,29 +95,31 @@ Make the animation definition available in your application by adding the reusab
 ### Styling the host and child components
 
-During a transition, a new view is inserted directly after the old one and both elements appear on screen at the same time. To prevent this, apply additional styling to the host view, and to the removed and inserted child views. The host view must use relative positioning, and the child views must use absolute positioning. Adding styling to the views animates the containers in place, without the  ...
+During a transition, a new view is inserted directly after the old one and both elements appear on screen at the same time. To prevent this behavior, update the host view to use relative positioning. Then, update the removed and inserted child views to use absolute positioning. Adding these styles to the views animates the containers in place and prevents one view from affecting the position of t ...
```
Nit: `of the other` sounds better to me (since we are talking about two views here:

```suggestion
During a transition, a new view is inserted directly after the old one and both elements appear on screen at the same time. To prevent this behavior, update the host view to use relative positioning. Then, update the removed and inserted child views to use absolute positioning. Adding these styles to the views animates the containers in place and prevents one view from affecting the position of the other on the page.
```

dario-piotrowicz replied:
Ah sorry I changed that as it made more sense to me :sweat:
I sort of got confused and thought it was a different example and there were more than 2 elements, sorry :sweat:

gkalpak reviewed (changes requested):
Almost there 🥁 :tada:

mary-poppins commented:
You can preview e685f94 at https://pr42885-e685f94.ngbuilds.io/.

dario-piotrowicz commented:
@gkalpak should be all good please have another look when you can :smile: 

By the way, just wanted to ask you, I'd like to rebase to master as I am a bit far behind I think, that would break the fixup hashes/links I use in comments, is that anything I can do about that? :thinking:

mary-poppins commented:
You can preview bdff258 at https://pr42885-bdff258.ngbuilds.io/.

gkalpak commented on aio/content/examples/animations/src/app/app.component.ts:
```diff
@@ -34,7 +33,7 @@ export class AppComponent {
 
 // #docregion prepare-router-outlet
   prepareRoute(outlet: RouterOutlet) {
-    return outlet && outlet.activatedRouteData && outlet.activatedRouteData.animation;
+    return outlet?.activatedRouteData?.animation;
```
I think it is fine.

gkalpak commented on aio/content/examples/animations/src/app/app.component.ts:
```diff
@@ -22,8 +22,10 @@ import { slideInAnimation } from './animations';
   animations: [
 // #enddocregion decorator
     slideInAnimation
+// #enddocregion toggle-app-animations, define
 // #docregion decorator
```
Nit: (Although either way works) I think the interleaved docregions would be a little easier to grok if these too lines were swapped:

```suggestion
// #docregion decorator
// #enddocregion toggle-app-animations, define
```

dario-piotrowicz replied:
yeah, sorry I must have gotten my wires crossed, this is of course much much better :sweat:

gkalpak reviewed (approved):
One last (really minor) suggestion. Otherwise looks perfect :100:

dario-piotrowicz commented:
> One last (really minor) suggestion. Otherwise looks perfect 100

Thank you so very much @gkalpak !!!! :smile: :smile: :smile:

mary-poppins commented:
You can preview 514e59d at https://pr42885-514e59d.ngbuilds.io/.

dario-piotrowicz committed:
fixup! docs(animations): improve the aio animations introduction

dario-piotrowicz committed:
docs(animations): improve the aio animations introduction

improve slightly the angular.io animation introduction guide, such
improvement are styling, information and clarity related

dario-piotrowicz committed:
fixup! docs(animations): improve the aio animations introduction

dario-piotrowicz committed:
fixup! docs(animations): improve the aio animations introduction

dario-piotrowicz committed:
fixup! docs(animations): improve the aio animations introduction

dario-piotrowicz committed:
fixup! docs(animations): improve the aio animations introduction

dario-piotrowicz committed:
fixup! docs(animations): improve the aio animations introduction

dario-piotrowicz committed:
fixup! docs(animations): improve the aio animations introduction

dario-piotrowicz commented:
@gkalpak thanks for rebasing for me! :slightly_smiling_face: 

sorry I'll try to be more vigilant next time :sweat_smile:

mary-poppins commented:
You can preview a1d476c at https://pr42885-a1d476c.ngbuilds.io/.
